openupgrade_scripts/scripts/pad/15.0.2.0/pre-migration.py

```diff
@@ -0,0 +1,41 @@
+from openupgradelib import openupgrade
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    """If all company have only one pad_server, this method will create configparam
+    If all company have more than one pad_server, user need to enter pad_server in settings
+    """
+    env.cr.execute(
+        "SELECT pad_server, pad_key FROM res_company WHERE pad_server IS NOT NULL AND pad_server IS NOT NULL"
+    )
+    all_company = env.cr.dictfetchall()
+    if all_company:
+        pad_config = {
+            "pad_server": all_company[0]["pad_server"],
+            "pad_key": all_company[0]["pad_key"],
+        }
+        for company in all_company:
+            if (
+                company.get("pad_server", False) != pad_config["pad_server"]
+                or company.get("pad_key", False) != pad_config["pad_key"]
+            ):
+                pad_config = {
+                    "pad_server": False,
+                    "pad_key": False,
+                }
+                break
+
+        if pad_config["pad_server"]:
+            env["ir.config_parameter"].create(
+                [
+                    {
+                        "key": "pad.pad_server",
+                        "value": pad_config["pad_server"],
+                    },
+                    {
+                        "key": "pad.pad_key",
+                        "value": pad_config["pad_key"],
+                    },
+                ]
+            )
```

Comment:
@daonamutc1 sử dụng `DISTINCT` để lấy kết quả không trùng nhau anh nhé
đang bị lỗi pre-commit nữa nhé
```suggestion
    env.cr.execute(
        "SELECT DISTINCT pad_server, pad_key FROM res_company WHERE pad_server IS NOT NULL AND pad_server IS NOT NULL"
    )
    pad_configs = env.cr.dictfetchall()
    if len(pad_configs) == 1:
            env["ir.config_parameter"].create(
                [
                    {
                        "key": "pad.pad_server",
                        "value": pad_configs[0]["pad_server"],
                    },
                    {
                        "key": "pad.pad_key",
                        "value": pad_configs[0]["pad_key"],
                    },
                ]
            )
```